zerver/lib/events.py

```diff
@@ -149,8 +149,8 @@ def fetch_initial_state_data(user_profile, event_types, queue_id,
 
     return state
 
-def apply_events(state, events, user_profile):
-    # type: (Dict[str, Any], Iterable[Dict[str, Any]], UserProfile) -> None
+def apply_events(state, events, user_profile, include_subscribers=True):
+    # type: (Dict[str, Any], Iterable[Dict[str, Any]], UserProfile, bool) -> None
     for event in events:
```

Comment:
We could use this pattern:

~~~ py
def apply_event():
    # super long if/elif

for event in events:
    apply_event()
~~~

And that would allow us to use "return" instead of "continue" later on down, which always feels so brittle.